exporter/aggregation_data.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package exporter
+
+import (
+	"math"
+)
+
+const epsilon = 1e-9
+
+// AggregationData is the aggregated data for exporters. Depending on the view
+// AggType, not all fields may be set.
+//
+// Most users won't directly access AggregationData, only exporters.
+type AggregationData struct {
+	Count           int64     // number of data points aggregated
+	Min             float64   // minimum value in the distribution
+	Max             float64   // max value in the distribution
+	Mean            float64   // mean of the distribution
+	SumOfSquaredDev float64   // sum of the squared deviation from the mean
+	CountPerBucket  []int64   // number of occurrences per bucket
+	Bounds          []float64 // histogram distribution of the values
+}
+
+// Equal compares two data points and returns true if they are mostly the same,
+// except for small floating point differences. Useful for testing.
+func (a AggregationData) Equal(a2 AggregationData) bool {
+	if len(a.CountPerBucket) != len(a2.CountPerBucket) {
+		return false
+	}
+	for i := range a.CountPerBucket {
```

Comment:
This is dangerous and can easily lead to bugs later on, could we make it
```go
for i := 0; i < len(a.CountPerBucket); i++ {
```

Reply:
i don't really understand this comment? do you mean to say one should never use this form of range?

Reply:
I mean that using
```go
for i := range a.CountPerBucket
```
is easier on the eyes.

For an `[]int*` slice could inadvertently lead to the use of i instead of `a.CountPerBucket[i]`. I'd use the former style for maps.

Reply:
This function was copied out of the stats/view package so I don't really want to change the body for now. I think the variable naming makes it clear that i is an index, not an element of the list.